### PR TITLE
Fixes #12472 : Migrate 'components/common' module and associated tests to TypeScript 

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Migrated following to typescript :
- [X] components/common/back_button.tsx
- [X] components/common/comment_icon.tsx
- [X] components/common/site_name_and_description.tsx
- [X] components/common/site_name_and_description.test.tsx
- [X] Type defination for OverlayTigger from React Bootstrap
 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12472
Jira https://mattermost.atlassian.net/browse/MM-18955

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5257,7 +5257,7 @@
         },
```
Comment (jespino) [inline package-lock.json]: The changes of this file are not related to the PR, do you mind to remove this changes from the PR?

Comment (M-ZubairAhmed) [reply package-lock.json]: done
